#!/usr/bin/env python3
# _*_ coding:utf-8 _*_
PACKAGE = "cv_06"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("threshold_maxval", int_t,    0, "threshold_maxval", 255, 0, 255)
gen.add("dilate_iterations",     int_t, 0, "dilate_iterations", 1, 0, 10)
gen.add("erode_iterations",     int_t, 0, "erode_iterations", 1, 0, 10)
gen.add("light_contour_area_assignment",     int_t, 0, "light_contour_area_assignment", 10, 1, 30)
gen.add("contour_angle_assignment",     int_t, 0, "contour_angle_assignment", 8, 1, 20)
gen.add("ratio_of_length_difference",    double_t, 0, "ratio_of_length_difference", 0.5, 0, 1.0)
gen.add("ratio_of_wideth_difference",    double_t, 0, "ratio_of_wideth_difference", 0.5, 0, 1.0)
gen.add("more_length_width_ratio",    double_t, 0, "more_length_width_ratio", 5.0, 0, 10.0)
gen.add("less_length_width_ratio",    double_t, 0, "less_length_width_ratio", 1.0, 0, 5.0)
gen.add("less_ratio_of_x_difference",    double_t,   0, "less_ratio_of_x_difference", 0.5,0,1.0 )
gen.add("more_ratio_of_y_difference",    double_t,   0, "more_ratio_of_y_difference", 2.0,1.0,3.0 )


exit(gen.generate(PACKAGE, "Sub", "dynamic"))